dragino trackerd

fzellini opened the pull request:
Dragino trackerd with meshtastic!

this pull include my previous ATH-10 sensors mods, however, for dragino, 
the only files are 

- src/gps/GPS.cpp
- variants/trackerd/*


![image](https://github.com/meshtastic/firmware/assets/8080238/8ed6757a-36e1-4e43-8ec7-274258270190)

alexander-smyslov commented on .gitmodules:
```diff
@@ -1,3 +1,3 @@
 [submodule "protobufs"]
 	path = protobufs
-	url = https://github.com/meshtastic/protobufs.git
```
I think need revert back to original protobufs

fzellini replied:
Done.
https://github.com/fzellini/meshtastic-firmware/blob/trackerd/.gitmodules

alexander-smyslov commented on src/mesh/generated/meshtastic/telemetry.pb.h:
```diff
@@ -57,7 +57,9 @@ typedef enum _meshtastic_TelemetrySensorType {
-    meshtastic_TelemetrySensorType_TSL25911FN = 22
+    meshtastic_TelemetrySensorType_TSL25911FN = 22,
+    /* AHT10/20 Temperature Sensor */
+    meshtastic_TelemetrySensorType_AHT10 = 23
```
need add AHT10 to protobuf like in https://github.com/meshtastic/protobufs/pull/499/files

fzellini replied:
OK, I don't understand in what manner: protobuf folder in my branch must be  included in branch an not in  submodule ?

alexander-smyslov replied:
I first we need add  AHT10 support to protobuf https://github.com/meshtastic/protobufs/pull/507/files and when use latest submodule  to use in this MR

fzellini replied:
> I first we need add AHT10 support to protobuf https://github.com/meshtastic/protobufs/pull/507/files and when use latest submodule to use in this MR

So i need to do a pull request in protobuf first ? Can't you enter it directly?

alexander-smyslov replied:
already done my me -   https://github.com/meshtastic/protobufs/pull/507/files

alexander-smyslov replied:
@fzellini can you add latest  protobuf submodule  to merge request?  https://github.com/meshtastic/protobufs/pull/507  merged

alexander-smyslov replied:
Just sync fzellini:trackerd brach with master (protobuf alredy merged https://github.com/meshtastic/firmware/pull/4003/files)

alexander-smyslov commented:
my opinion:  add ath10 support https://github.com/meshtastic/firmware/pull/3977/files and when add dragino trackerd  in this merge request

thebentern commented on src/gps/GPS.cpp:
```diff
@@ -1321,6 +1321,14 @@ GPS *GPS::createGps()
+         * https://www.waveshare.com/wiki/L76K_GPS_Module
+         */
+
+        _serial_gps->updateBaudRate(9600);
```
We default to 9600 anyway. This is not needed

thebentern replied:
This is still not needed

fzellini replied:
reverted back

thebentern commented on variants/trackerd/platformio-custom.py:
```diff
@@ -0,0 +1,77 @@
+# trunk-ignore-all(ruff/F821)
+# trunk-ignore-all(flake8/F821): For SConstruct imports
+import sys
```
I'm not sure why a custom platformio script would be needed for this board? It's a standard esp32 target

fzellini replied:
with bin/platformio-custom.py the dragino goes in bootloop.

_Guru Meditation Error: Core  0 panic'ed (Cache disabled but cached memory region accessed).
Backtrace: 0x402097d5:0x3ffe3b10 0x4008284b:0x3ffe3b50 0x4013199d:0x3ffe3b80 0x40082941:0x3ffe3ba0 0x401319ff:0x3ffe3bd0 0x401361f0:0x3ffe3c20 0x400839dd:0x3ffe3c50 0x40079306:0x3ffe3c90
Guru Meditation Error: Core  0 panic'ed (Cache disabled but cached memory region accessed)._

my version fix the problem
I spend 2 days to understand why, i suspect there is something wrong with the settings provided by the python script.
In order to not touch original script, i rewrite it in variants.
Dragino has esp32 pico D4, maybe slighty different from other esp32 ?

![pcb-photo](https://github.com/meshtastic/firmware/assets/8080238/7e2b1fa9-6fde-425c-a556-ecc950763f19)

alexander-smyslov replied:
check without extra_scripts = variants/trackerd/platformio-custom.py - work well 
before also have  bootloop

alexander-smyslov replied:
@fzellini  can you check your build with latest code without extra_scripts ?

thebentern reviewed (changes requested):
Please address the feedback. Some of these changes are not necessary

alexander-smyslov commented:
check without extra_scripts = variants/trackerd/platformio-custom.py - work well
before also have bootloop
@fzellini can you check your build with latest code without extra_scripts ?

fzellini commented:
> check without extra_scripts = variants/trackerd/platformio-custom.py - work well before also have bootloop @fzellini can you check your build with latest code without extra_scripts ?
OK works, in the last days we are working on same issue.
I have reverted trackerd/platformio.ini and removed custom script
